cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/r4/R4MeasureDefBuilder.java

```diff
@@ -262,27 +239,102 @@ private void validateImprovementNotationCode(CodeableConcept improvementNotation
     }
 
-    private boolean isGroupIncreaseImprovementNotation(Measure measure, MeasureGroupComponent group) {
-        // default improvement Notation
-        boolean isIncreaseImpNotation = true;
-        boolean useGroupImpNotation = groupHasImprovementNotationExt(group);
-        if (useGroupImpNotation) {
-            isIncreaseImpNotation = isIncreaseImprovementNotation(getGroupImprovementNotationExt(group));
-        } else if (measure.hasImprovementNotation()) {
-            isIncreaseImpNotation = isIncreaseImprovementNotation(measure.getImprovementNotation());
+    public CodeDef getMeasureImprovementNotation(Measure measure) {
+        if (measure.hasImprovementNotation()) {
+            var improvementNotationValue = measure.getImprovementNotation();
+            var codeDef = new CodeDef(
+                    improvementNotationValue.getCodingFirstRep().getSystem(),
+                    improvementNotationValue.getCodingFirstRep().getCode());
+            validateImprovementNotationCode(codeDef);
+            return codeDef;
         }
+        return null;
+    }
+
+    public CodeDef getGroupImpNotation(MeasureGroupComponent group) {
+        var ext = group.getExtensionByUrl(MEASUREREPORT_IMPROVEMENT_NOTATION_EXTENSION);
+        if (ext != null) {
+            var value = ext.getValue();
+            assert value instanceof CodeableConcept;
+            CodeableConcept coding = (CodeableConcept) value;
+            var codeDef = new CodeDef(
+                    coding.getCodingFirstRep().getSystem(),
+                    coding.getCodingFirstRep().getCode());
+            validateImprovementNotationCode(codeDef);
+            return codeDef;
+        }
+        return null;
+    }
 
-        return isIncreaseImpNotation;
+    public MeasureScoring getGroupMeasureScoring(MeasureGroupComponent group) {
+        var ext = group.getExtensionByUrl(CQFM_SCORING_EXT_URL);
+        if (ext != null) {
+            var extVal = ext.getValue();
+            assert extVal instanceof CodeableConcept;
+            CodeableConcept coding = (CodeableConcept) extVal;
+            return getMeasureScoring(coding.getCodingFirstRep().getCode());
+        }
+        return null;
+    }
+
+    public CodeDef getGroupPopulationBasis(MeasureGroupComponent group) {
+        var ext = group.getExtensionByUrl(MeasureConstants.POPULATION_BASIS_URL);
+        // check for population-basis Extension, assume boolean if no Extension is found
+        if (ext != null) {
+            return makeCodeDefFromExtension(ext);
+        }
+        return null;
+    }
+
+    private MeasureScoring getScoringDef(MeasureScoring measureScoring, MeasureScoring groupScoring) {
+        if (groupScoring == null && measureScoring == null) {
+            throw new IllegalArgumentException("MeasureScoring must be specified on Group or Measure");
+        }
+        if (groupScoring != null) {
+            return groupScoring;
+        } else {
```

Comment:
Nitpick:  return short-circuits execution of if, so you don't need else to follow it, just return after the if block.